File: packages/test-pkg/ios/TestPkgModule.swift
import ExpoModulesCore

public class TestPkgModule: Module {
  public func definition() -> ModuleDefinition {
    name("TestPkg")

    function("helloAsync") { (options: [String: String]) in
      print("Hello 👋")
    }

    viewManager {
      view {
        TestPkgView()
      }

      prop("name") { (view: TestPkgView, prop: Int) in
        print(prop)
      }
    }
  }
}

File: packages/test-pkg/android/src/main/java/expo/modules.testpkg/TestPkgModule.kt
package expo.modules.testpkg

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class TestPkgModule : Module() {
  override fun definition() = ModuleDefinition {
    name("TestPkg")

    function("helloAsync") { options: Map<String, String> ->
      println("Hello 👋")
    }

    viewManager {
      view { context -> 
        TestPkgView(context) 
      }

      prop("name") { view: TestPkgView, prop: Int ->
        println(prop)
      }
    }
  }
}
